File: src/components/core/Container/platforms/ios/TokenMapping.swift
/**
 * Token-to-SwiftUI Mapping Functions
 * 
 * Converts Container component props to SwiftUI values that reference
 * design system tokens. These functions handle the translation from platform-agnostic
 * token names to iOS-specific SwiftUI types (EdgeInsets, Color, CGFloat, etc.).
 * 
 * Token Resolution:
 * - Input: Token name (e.g., 'color.primary', 'shadow.container')
 * - Output: SwiftUI type (Color, CGFloat, EdgeInsets, etc.)
 * 
 * Note: This file contains placeholder implementations that will be replaced
 * by actual token lookups from generated Swift constants when the token
 * generation system is complete.
 * 
 * @see .kiro/specs/010-container-component/design.md for token consumption strategy
 * @see Requirements 2.1-2.5, 3.1-3.3, 4.1-4.3, 5.1-5.3, 6.1-6.3, 7.1-7.3, 8.1-8.4, 9.1-9.6
 */

import SwiftUI

// MARK: - Padding Mapping

/**
 * Map padding value to SwiftUI EdgeInsets
 * 
 * Converts padding prop value to EdgeInsets using space.inset tokens.
 * Returns EdgeInsets() (zero padding) for .none value.
 * 
 * @param padding - Padding prop value
 * @returns SwiftUI EdgeInsets
 * 
 * @example
 * ```swift
 * mapPaddingToEdgeInsets(.p200) // Returns EdgeInsets with 16pt on all sides
 * mapPaddingToEdgeInsets(.none) // Returns EdgeInsets() (zero padding)
 * ```
 * 
 * @see Requirements 2.1, 3.1-3.7
 */
func mapPaddingToEdgeInsets(_ padding: PaddingValue) -> EdgeInsets {
    switch padding {
    case .none:
        return EdgeInsets()
    case .p050:
        return EdgeInsets(top: spaceInset050, leading: spaceInset050, bottom: spaceInset050, trailing: spaceInset050)
    case .p100:
        return EdgeInsets(top: spaceInset100, leading: spaceInset100, bottom: spaceInset100, trailing: spaceInset100)
    case .p150:
        return EdgeInsets(top: spaceInset150, leading: spaceInset150, bottom: spaceInset150, trailing: spaceInset150)
    case .p200:
        return EdgeInsets(top: spaceInset200, leading: spaceInset200, bottom: spaceInset200, trailing: spaceInset200)
    case .p300:
        return EdgeInsets(top: spaceInset300, leading: spaceInset300, bottom: spaceInset300, trailing: spaceInset300)
    case .p400:
        return EdgeInsets(top: spaceInset400, leading: spaceInset400, bottom: spaceInset400, trailing: spaceInset400)
    }
}

// MARK: - Border Mapping

/**
 * Map border value to SwiftUI line width
 * 
 * Converts border prop value to CGFloat using border width tokens.
 * Returns 0 for .none value.
 * 
 * @param border - Border prop value
 * @returns Border width as CGFloat
 * 
 * @example
 * ```swift
 * mapBorderToLineWidth(.default) // Returns 1.0 (1pt)
 * mapBorderToLineWidth(.none) // Returns 0
 * ```
 * 
 * @see Requirements 2.4, 6.1-6.5
 */
func mapBorderToLineWidth(_ border: BorderValue) -> CGFloat {
    switch border {
    case .none:
        return 0
    case .default:
        return borderDefault
    case .emphasis:
        return borderEmphasis
    case .heavy:
        return borderHeavy
    }
}

/**
 * Get border color from token
 * 
 * Returns the color.border token value for border styling.
 * This is a constant color used for all borders.
 * 
 * @returns Border color as SwiftUI Color
 * 
 * @example
 * ```swift
 * getBorderColor() // Returns color.border token value
 * ```
 * 
 * @see Requirements 6.5
 */
func getBorderColor() -> Color {
    return colorBorder
}

// MARK: - Border Radius Mapping

/**
 * Map border radius value to SwiftUI corner radius
 * 
 * Converts borderRadius prop value to CGFloat using radius tokens.
 * Returns 0 for .none value.
 * 
 * @param borderRadius - Border radius prop value
 * @returns Corner radius as CGFloat
 * 
 * @example
 * ```swift
 * mapBorderRadiusToCornerRadius(.normal) // Returns 8.0 (8pt)
 * mapBorderRadiusToCornerRadius(.none) // Returns 0
 * ```
 * 
 * @see Requirements 2.5, 7.1-7.4
 */
func mapBorderRadiusToCornerRadius(_ borderRadius: BorderRadiusValue) -> CGFloat {
    switch borderRadius {
    case .none:
        return 0
    case .tight:
        return radius050
    case .normal:
        return radius100
    case .loose:
        return radius200
    }
}

// MARK: - Color Mapping

/**
 * Resolve color token name to SwiftUI Color
 * 
 * Converts color token name to SwiftUI Color.
 * Returns Color.clear if token name is nil or empty.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName - Color token name (e.g., "color.primary")
 * @returns SwiftUI Color
 * 
 * @example
 * ```swift
 * resolveColorToken("color.primary") // Returns primary color
 * resolveColorToken("color.surface") // Returns surface color
 * resolveColorToken(nil) // Returns Color.clear
 * ```
 * 
 * @see Requirements 2.2, 4.1-4.4
 */
func resolveColorToken(_ tokenName: String?) -> Color {
    guard let tokenName = tokenName, !tokenName.isEmpty else {
        return Color.clear
    }
    
    // TODO: Implement token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Swift constants
    // For now, return a placeholder color
    
    // Example of what the generated code might look like:
    // switch tokenName {
    // case "color.primary":
    //     return colorPrimary
    // case "color.surface":
    //     return colorSurface
    // case "color.background":
    //     return colorBackground
    // default:
    //     return Color.clear
    // }
    
    return Color.blue // Placeholder
}

// MARK: - Shadow Mapping

/**
 * Shadow properties structure
 * 
 * Encapsulates all shadow properties needed for SwiftUI shadow modifier.
 */
struct ShadowProperties {
    let color: Color
    let radius: CGFloat
    let x: CGFloat
    let y: CGFloat
}

/**
 * Resolve shadow token to shadow properties
 * 
 * Converts shadow token name to ShadowProperties containing color, radius, and offsets.
 * Returns zero shadow (clear color, 0 radius) if token name is nil or empty.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName - Shadow token name (e.g., "shadow.container")
 * @returns ShadowProperties with color, radius, x, and y
 * 
 * @example
 * ```swift
 * let shadow = resolveShadowToken("shadow.container")
 * // Returns ShadowProperties with appropriate values
 * 
 * let noShadow = resolveShadowToken(nil)
 * // Returns ShadowProperties(color: .clear, radius: 0, x: 0, y: 0)
 * ```
 * 
 * @see Requirements 2.3, 5.1-5.4
 */
func resolveShadowToken(_ tokenName: String?) -> ShadowProperties {
    guard let tokenName = tokenName, !tokenName.isEmpty else {
        return ShadowProperties(color: Color.clear, radius: 0, x: 0, y: 0)
    }
    
    // TODO: Implement shadow token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Swift constants
    // For now, return placeholder shadow properties
    
    // Example of what the generated code might look like:
    // switch tokenName {
    // case "shadow.container":
    //     return ShadowProperties(
    //         color: shadowContainerColor,
    //         radius: shadowContainerRadius,
    //         x: shadowContainerX,
    //         y: shadowContainerY
    //     )
    // case "shadow.modal":
    //     return ShadowProperties(
    //         color: shadowModalColor,
    //         radius: shadowModalRadius,
    //         x: shadowModalX,
    //         y: shadowModalY
    //     )
    // default:
    //     return ShadowProperties(color: .clear, radius: 0, x: 0, y: 0)
    // }
    
    // Placeholder shadow
    return ShadowProperties(
        color: Color.black.opacity(0.1),
        radius: 8,
        x: 0,
        y: 4
    )
}

// MARK: - Opacity Mapping

/**
 * Resolve opacity token name to Double
 * 
 * Converts opacity token name to Double value (0.0 to 1.0).
 * Returns 1.0 (fully opaque) if token name is nil or empty.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName - Opacity token name (e.g., "opacity.subtle")
 * @returns Opacity value as Double (0.0 to 1.0)
 * 
 * @example
 * ```swift
 * resolveOpacityToken("opacity.subtle") // Returns 0.9
 * resolveOpacityToken("opacity.ghost") // Returns 0.3
 * resolveOpacityToken(nil) // Returns 1.0
 * ```
 * 
 * @see Requirements 8.1-8.4
 */
func resolveOpacityToken(_ tokenName: String?) -> Double {
    guard let tokenName = tokenName, !tokenName.isEmpty else {
        return 1.0
    }
    
    // TODO: Implement opacity token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Swift constants
    // For now, return placeholder opacity
    
    // Example of what the generated code might look like:
    // switch tokenName {
    // case "opacity.subtle":
    //     return opacitySubtle
    // case "opacity.medium":
    //     return opacityMedium
    // case "opacity.heavy":
    //     return opacityHeavy
    // case "opacity.ghost":
    //     return opacityGhost
    // default:
    //     return 1.0
    // }
    
    return 0.9 // Placeholder
}

// MARK: - Layering Mapping

/**
 * Map layering value to SwiftUI z-index
 * 
 * Converts layering prop value to Double for z-index using z-index tokens (iOS platform).
 * Returns 0 if layering is nil.
 * 
 * Note: iOS uses z-index tokens for stacking order (same as web).
 * Android uses elevation tokens which handle both stacking and shadow.
 * 
 * @param layering - Layering prop value
 * @returns Z-index value as Double
 * 
 * @example
 * ```swift
 * mapLayeringToZIndex(.modal) // Returns 400.0
 * mapLayeringToZIndex(.navigation) // Returns 200.0
 * mapLayeringToZIndex(nil) // Returns 0
 * ```
 * 
 * @see Requirements 9.1-9.9
 */
func mapLayeringToZIndex(_ layering: LayeringValue?) -> Double {
    guard let layering = layering else {
        return 0
    }
    
    switch layering {
    case .container:
        return Double(zIndexContainer)
    case .navigation:
        return Double(zIndexNavigation)
    case .dropdown:
        return Double(zIndexDropdown)
    case .modal:
        return Double(zIndexModal)
    case .toast:
        return Double(zIndexToast)
    case .tooltip:
        return Double(zIndexTooltip)
    }
}

// MARK: - Token Constants (Placeholders)

/**
 * Token constants
 * 
 * These are placeholders that will be replaced by generated token constants
 * from the build system. The actual values will come from:
 * - Space tokens: src/tokens/SpacingTokens.ts
 * - Border tokens: src/tokens/BorderTokens.ts
 * - Radius tokens: src/tokens/RadiusTokens.ts
 * - Z-index tokens: src/tokens/semantic/LayeringTokens.ts
 * - Color tokens: src/tokens/semantic/ColorTokens.ts
 * - Opacity tokens: src/tokens/semantic/OpacityTokens.ts
 * - Shadow tokens: src/tokens/semantic/ShadowTokens.ts
 */

// Space.inset tokens (padding)
private let spaceInset050: CGFloat = 4   // 0.5 × base
private let spaceInset100: CGFloat = 8   // 1 × base
private let spaceInset150: CGFloat = 12  // 1.5 × base
private let spaceInset200: CGFloat = 16  // 2 × base
private let spaceInset300: CGFloat = 24  // 3 × base
private let spaceInset400: CGFloat = 32  // 4 × base

// Border tokens
private let borderDefault: CGFloat = 1   // 1pt
private let borderEmphasis: CGFloat = 2  // 2pt
private let borderHeavy: CGFloat = 4     // 4pt

// Radius tokens
private let radius050: CGFloat = 4   // 4pt
private let radius100: CGFloat = 8   // 8pt
private let radius200: CGFloat = 16  // 16pt

// Z-index tokens (layering)
private let zIndexContainer: Int = 100
private let zIndexNavigation: Int = 200
private let zIndexDropdown: Int = 300
private let zIndexModal: Int = 400
private let zIndexToast: Int = 500
private let zIndexTooltip: Int = 600

// Color tokens
private let colorBorder: Color = Color.gray.opacity(0.3)

File: src/components/core/Container/platforms/android/TokenMapping.kt
/**
 * Token-to-Compose Mapping Functions
 * 
 * Converts Container component props to Jetpack Compose values that reference
 * design system tokens. These functions handle the translation from platform-agnostic
 * token names to Android-specific Compose types (Dp, Color, Shape, etc.).
 * 
 * Token Resolution:
 * - Input: Token name (e.g., 'color.primary', 'shadow.container')
 * - Output: Compose type (Color, Dp, Shape, etc.)
 * 
 * Note: This file contains placeholder implementations that will be replaced
 * by actual token lookups from generated Kotlin constants when the token
 * generation system is complete.
 * 
 * @see .kiro/specs/010-container-component/design.md for token consumption strategy
 * @see Requirements 2.1-2.5, 3.1-3.3, 4.1-4.3, 5.1-5.3, 6.1-6.3, 7.1-7.3, 8.1-8.4, 9.1-9.6
 */

package com.designerpunk.components.core

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// MARK: - Padding Mapping

/**
 * Map padding value to Compose Dp
 * 
 * Converts padding prop value to Dp using space.inset tokens.
 * Returns 0.dp for PaddingValue.None.
 * 
 * @param padding Padding prop value
 * @return Dp value for padding
 * 
 * @example
 * ```kotlin
 * mapPaddingToDp(PaddingValue.P200) // Returns 16.dp
 * mapPaddingToDp(PaddingValue.None) // Returns 0.dp
 * ```
 * 
 * @see Requirements 2.1, 3.1-3.7
 */
fun mapPaddingToDp(padding: PaddingValue): Dp {
    return when (padding) {
        PaddingValue.None -> 0.dp
        PaddingValue.P050 -> spaceInset050
        PaddingValue.P100 -> spaceInset100
        PaddingValue.P150 -> spaceInset150
        PaddingValue.P200 -> spaceInset200
        PaddingValue.P300 -> spaceInset300
        PaddingValue.P400 -> spaceInset400
    }
}

// MARK: - Border Mapping

/**
 * Map border value to Compose Dp width
 * 
 * Converts border prop value to Dp using border width tokens.
 * Returns 0.dp for BorderValue.None.
 * 
 * @param border Border prop value
 * @return Border width as Dp
 * 
 * @example
 * ```kotlin
 * mapBorderToWidth(BorderValue.Default) // Returns 1.dp
 * mapBorderToWidth(BorderValue.None) // Returns 0.dp
 * ```
 * 
 * @see Requirements 2.4, 6.1-6.5
 */
fun mapBorderToWidth(border: BorderValue): Dp {
    return when (border) {
        BorderValue.None -> 0.dp
        BorderValue.Default -> borderDefault
        BorderValue.Emphasis -> borderEmphasis
        BorderValue.Heavy -> borderHeavy
    }
}

/**
 * Get border color from token
 * 
 * Returns the color.border token value for border styling.
 * This is a constant color used for all borders.
 * 
 * @return Border color as Compose Color
 * 
 * @example
 * ```kotlin
 * getBorderColor() // Returns color.border token value
 * ```
 * 
 * @see Requirements 6.5
 */
fun getBorderColor(): Color {
    return colorBorder
}

// MARK: - Border Radius Mapping

/**
 * Get RoundedCornerShape from border radius value
 * 
 * Converts borderRadius prop value to RoundedCornerShape using radius tokens.
 * Returns RoundedCornerShape(0.dp) for BorderRadiusValue.None.
 * 
 * @param borderRadius Border radius prop value
 * @return RoundedCornerShape for the border radius
 * 
 * @example
 * ```kotlin
 * getRoundedCornerShape(BorderRadiusValue.Normal) // Returns RoundedCornerShape(8.dp)
 * getRoundedCornerShape(BorderRadiusValue.None) // Returns RoundedCornerShape(0.dp)
 * ```
 * 
 * @see Requirements 2.5, 7.1-7.4
 */
fun getRoundedCornerShape(borderRadius: BorderRadiusValue): RoundedCornerShape {
    val radius = when (borderRadius) {
        BorderRadiusValue.None -> 0.dp
        BorderRadiusValue.Tight -> radius050
        BorderRadiusValue.Normal -> radius100
        BorderRadiusValue.Loose -> radius200
    }
    return RoundedCornerShape(radius)
}

// MARK: - Color Mapping

/**
 * Resolve color token name to Compose Color
 * 
 * Converts color token name to Compose Color.
 * Returns Color.Transparent if token name is null or empty.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName Color token name (e.g., "color.primary")
 * @return Compose Color
 * 
 * @example
 * ```kotlin
 * resolveColorToken("color.primary") // Returns primary color
 * resolveColorToken("color.surface") // Returns surface color
 * resolveColorToken(null) // Returns Color.Transparent
 * ```
 * 
 * @see Requirements 2.2, 4.1-4.4
 */
fun resolveColorToken(tokenName: String?): Color {
    if (tokenName.isNullOrEmpty()) {
        return Color.Transparent
    }
    
    // TODO: Implement token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Kotlin constants
    // For now, return a placeholder color
    
    // Example of what the generated code might look like:
    // return when (tokenName) {
    //     "color.primary" -> colorPrimary
    //     "color.surface" -> colorSurface
    //     "color.background" -> colorBackground
    //     else -> Color.Transparent
    // }
    
    return Color.Blue // Placeholder
}

// MARK: - Shadow Mapping

/**
 * Resolve shadow token to elevation Dp
 * 
 * Converts shadow token name to Compose elevation Dp.
 * Returns 0.dp if token name is null or empty.
 * 
 * On Android, elevation handles both stacking order and shadow rendering.
 * This function maps shadow tokens to appropriate elevation values.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName Shadow token name (e.g., "shadow.container")
 * @return Elevation value as Dp
 * 
 * @example
 * ```kotlin
 * mapShadowToElevation("shadow.container") // Returns 8.dp
 * mapShadowToElevation("shadow.modal") // Returns 16.dp
 * mapShadowToElevation(null) // Returns 0.dp
 * ```
 * 
 * @see Requirements 2.3, 5.1-5.4
 */
fun mapShadowToElevation(tokenName: String?): Dp {
    if (tokenName.isNullOrEmpty()) {
        return 0.dp
    }
    
    // TODO: Implement shadow token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Kotlin constants
    // For now, return placeholder elevation based on token name patterns
    
    // Example of what the generated code might look like:
    // return when (tokenName) {
    //     "shadow.sunrise" -> shadowSunriseElevation
    //     "shadow.noon" -> shadowNoonElevation
    //     "shadow.dusk" -> shadowDuskElevation
    //     "shadow.container" -> shadowContainerElevation
    //     "shadow.modal" -> shadowModalElevation
    //     else -> 4.dp
    // }
    
    // Placeholder mapping based on token name patterns
    return when {
        tokenName.contains("sunrise") -> 2.dp
        tokenName.contains("noon") -> 4.dp
        tokenName.contains("dusk") -> 8.dp
        tokenName.contains("container") -> 8.dp
        tokenName.contains("navigation") -> 4.dp
        tokenName.contains("dropdown") -> 8.dp
        tokenName.contains("modal") -> 16.dp
        else -> 4.dp  // Default elevation
    }
}

// MARK: - Opacity Mapping

/**
 * Resolve opacity token name to Float
 * 
 * Converts opacity token name to Float value (0.0 to 1.0).
 * Returns 1.0f (fully opaque) if token name is null or empty.
 * 
 * Note: This is a placeholder implementation. The actual implementation
 * will use generated token constants from the build system.
 * 
 * @param tokenName Opacity token name (e.g., "opacity.subtle")
 * @return Opacity value as Float (0.0 to 1.0)
 * 
 * @example
 * ```kotlin
 * resolveOpacityToken("opacity.subtle") // Returns 0.9f
 * resolveOpacityToken("opacity.ghost") // Returns 0.3f
 * resolveOpacityToken(null) // Returns 1.0f
 * ```
 * 
 * @see Requirements 8.1-8.4
 */
fun resolveOpacityToken(tokenName: String?): Float {
    if (tokenName.isNullOrEmpty()) {
        return 1.0f
    }
    
    // TODO: Implement opacity token resolution via generated token constants
    // This will be replaced by actual token lookup from generated Kotlin constants
    // For now, return placeholder opacity
    
    // Example of what the generated code might look like:
    // return when (tokenName) {
    //     "opacity.subtle" -> opacitySubtle
    //     "opacity.medium" -> opacityMedium
    //     "opacity.heavy" -> opacityHeavy
    //     "opacity.ghost" -> opacityGhost
    //     else -> 1.0f
    // }
    
    return 0.9f // Placeholder
}

// MARK: - Layering Mapping

/**
 * Map layering value to elevation Dp
 * 
 * Converts layering prop value to Compose elevation Dp using elevation tokens.
 * Returns 0.dp if layering is null.
 * 
 * On Android, elevation tokens handle both stacking order and shadow rendering,
 * following Material Design guidelines.
 * 
 * @param layering Layering prop value
 * @return Elevation value as Dp
 * 
 * @example
 * ```kotlin
 * mapLayeringToElevation(LayeringValue.Modal) // Returns 16.dp
 * mapLayeringToElevation(LayeringValue.Navigation) // Returns 4.dp
 * mapLayeringToElevation(null) // Returns 0.dp
 * ```
 * 
 * @see Requirements 9.1-9.9
 */
fun mapLayeringToElevation(layering: LayeringValue?): Dp {
    return when (layering) {
        null -> 0.dp
        LayeringValue.Container -> elevationContainer
        LayeringValue.Navigation -> elevationNavigation
        LayeringValue.Dropdown -> elevationDropdown
        LayeringValue.Modal -> elevationModal
        LayeringValue.Toast -> elevationToast
        LayeringValue.Tooltip -> elevationTooltip
    }
}

// MARK: - Token Constants (Placeholders)

/**
 * Token constants
 * 
 * These are placeholders that will be replaced by generated token constants
 * from the build system. The actual values will come from:
 * - Space tokens: src/tokens/SpacingTokens.ts
 * - Border tokens: src/tokens/BorderTokens.ts
 * - Radius tokens: src/tokens/RadiusTokens.ts
 * - Elevation tokens: src/tokens/semantic/LayeringTokens.ts (Android-specific)
 * - Color tokens: src/tokens/semantic/ColorTokens.ts
 * - Opacity tokens: src/tokens/semantic/OpacityTokens.ts
 * - Shadow tokens: src/tokens/semantic/ShadowTokens.ts
 */

// Import DesignTokens for token references
import com.designerpunk.tokens.DesignTokens

// Space.inset tokens (padding)
private val spaceInset050: Dp = DesignTokens.space_inset_050.dp
private val spaceInset100: Dp = DesignTokens.space_inset_100.dp
private val spaceInset150: Dp = DesignTokens.space_inset_150.dp
private val spaceInset200: Dp = DesignTokens.space_inset_200.dp
private val spaceInset300: Dp = DesignTokens.space_inset_300.dp
private val spaceInset400: Dp = DesignTokens.space_inset_400.dp

// Border tokens
private val borderDefault: Dp = DesignTokens.border_default.dp
private val borderEmphasis: Dp = DesignTokens.border_emphasis.dp
private val borderHeavy: Dp = DesignTokens.border_heavy.dp

// Radius tokens
private val radius050: Dp = DesignTokens.radius_050.dp
private val radius100: Dp = DesignTokens.radius_100.dp
private val radius200: Dp = DesignTokens.radius_200.dp

// Elevation tokens (layering - Android-specific)
// On Android, elevation handles both stacking order and shadow rendering
private val elevationContainer: Dp = DesignTokens.elevation_container.dp
private val elevationNavigation: Dp = DesignTokens.elevation_navigation.dp
private val elevationDropdown: Dp = DesignTokens.elevation_dropdown.dp
private val elevationModal: Dp = DesignTokens.elevation_modal.dp
private val elevationToast: Dp = DesignTokens.elevation_toast.dp
private val elevationTooltip: Dp = DesignTokens.elevation_tooltip.dp

// Color tokens
private val colorBorder: Color = Color(DesignTokens.color_border)
